sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ProjectingInternalRow.scala

```diff
@@ -19,16 +19,14 @@ package org.apache.spark.sql.catalyst
 
 import org.apache.spark.SparkUnsupportedOperationException
 import org.apache.spark.sql.catalyst.util.{ArrayData, MapData}
-import org.apache.spark.sql.types.{DataType, Decimal, StructType}
+import org.apache.spark.sql.types.{DataType, Decimal}
 import org.apache.spark.unsafe.types.{CalendarInterval, UTF8String, VariantVal}
 
 /**
  * An [[InternalRow]] that projects particular columns from another [[InternalRow]] without copying
  * the underlying data.
  */
-case class ProjectingInternalRow(schema: StructType,
```

Comment:
When I added this class, I was using `schema` in input validation. That said, it was a fairly useless check. I've reworked input validation in this PR and we no longer need `schema` here. This class is internal and only used in DML. I have to problem keeping it as before but it is kind of useless now.